aepsych/models/semi_p.py

```diff
@@ -375,7 +376,9 @@ def predict(
         m, v = samps.mean(0), samps.var(0)
         return promote_0d(m), promote_0d(v)
 
-    def posterior(self, X, posterior_transform=None):
+    def posterior(self, X: Union[torch.Tensor, np.ndarray], posterior_transform: Optional[PosteriorTransform] = None) -> SemiPPosterior:
+        if isinstance(X, np.ndarray):
```

Comment:
Is there a reason we still need to support numpy arrays in this method, or should we only allow tensors?

Reply:
You're right; I found the root of the issue in `GPClassificationModel`, as it still supports both. I’ve fixed it where needed.